File: solutions/go/darts/1/darts.go
package darts

import "math"

func Score(x, y float64) int {
	dartLocation := math.Sqrt(x*x + y*y)
    switch {
    case dartLocation <= 1.0:
        return 10
    case dartLocation <= 5.0:
    	return 5
    case dartLocation <= 10.0:
    	return 1
    default:
    	return 0
    }
}

File: solutions/cpp/darts/1/darts.cpp
#include "darts.h"

namespace darts {

int score(double x, double y) {
    double distance_squared = x * x + y * y;

    if (distance_squared <= 1.0) {      
        return 10;
    } else if (distance_squared <= 25.0) {
        return 5;
    } else if (distance_squared <= 100.0) {
        return 1;
    }
    return 0;
}

}  // namespace darts